Library/Homebrew/keg.rb

```diff
@@ -456,22 +455,21 @@ def oldname_opt_record
     end
   end
 
-  # TODO: refactor to use keyword arguments.
-  def link(**options)
+  def link(verbose: false, dry_run: false, overwrite: false)
     raise AlreadyLinkedError, self if linked_keg_record.directory?
 
     ObserverPathnameExtension.reset_counts!
 
-    optlink(**options) unless options[:dry_run]
+    optlink(verbose: verbose, dry_run: dry_run, overwrite: overwrite) unless dry_run
 
     # yeah indeed, you have to force anything you need in the main tree into
     # these dirs REMEMBER that *NOT* everything needs to be in the main tree
-    link_dir("etc", **options) { :mkpath }
-    link_dir("bin", **options) { :skip_dir }
-    link_dir("sbin", **options) { :skip_dir }
-    link_dir("include", **options) { :link }
+    link_dir("etc", verbose: verbose, dry_run: dry_run, overwrite: overwrite) { :mkpath }
```

Comment:
I wonder if we could/should set `@verbose, `@dry_run`, `@overwrite` and read from them in e.g. `link_dir` (or any `private`/`protected` method) to share passing these through every time. There's only actually 3 public methods that take these `**options`. Any thoughts @reitermarkus?

Reply:
No, we should pass them along, since this decouples the options from the `Keg` instance and in general makes it easier to reason about where these options are coming from.

Reply:
Fine with me.